django/db/models/expressions.py

```diff
@@ -105,14 +105,21 @@ def bitleftshift(self, other):
     def bitrightshift(self, other):
         return self._combine(other, self.BITRIGHTSHIFT, False)
 
+    def __xor__(self, other):
+        if getattr(self, 'conditional', False) and getattr(other, 'conditional', False):
+            return Q(self) & Q(other)
```

Comment:
Looks like you forgot to change the operator:

```suggestion
            return Q(self) ^ Q(other)
```

Reply:
Eeeek, good catch, looks like none of my unit tests hit that part. I'm not sure if Combinable/Expression/Subquery/Exists has an appropriate place for unit tests like, but I'm moving around some decorators and adding one.

If you have an idea for a better spot for `test_logical_xor_subquery`, let me know